scripts/zones/Windurst_Walls/npcs/Shantotto.lua

```diff
@@ -179,6 +189,15 @@ function onEventFinish(player,csid,option)
         player:setVar("MissionStatus",8)
     elseif (csid == 399) then
         player:setVar("ShantottoCS",0)
+    elseif csid == 506 then
+        player:completeMission(AMK,dsp.mission.id.amk.CURSES_A_HORRIFICALLY_HARROWING_HEX)
+        player:addMission(AMK,dsp.mission.id.amk.AN_ERRAND_THE_PROFESSORS_PRICE)
+    elseif csid == 507 and option == 0 then -- Dont Pay
+        player:needToZone(true)
+    elseif csid == 507 and option == 1 then -- Pay
+        player:delGil(5000)
```

Comment:
Make sure you check greater than 5000 gil here before you delete it (if "gil in param" above).